### PR TITLE
fix(api): add timeout and better logging to Horizon Gateway

### PR DESCRIPTION
**Pull request by AlexMoulin**
## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/api/src/server/utils/horizon-gateway.js [hunk 1]
```diff
@@ -107,19 +107,29 @@ export const getAppealFromHorizon = async (caseReference) => {
...
+			}
 		})
 		.json()
 		.catch((error) => {
-			if (error.response.body && JSON.stringify(error.response.body).includes('faultstring')) {
+			if (error.code === 'ETIMEDOUT') {
+				logger.error('Call to Horizon timed-out');
+			} else if (
+				error.response.body &&
+				JSON.stringify(error.response.body).includes('faultstring')
```
Comment (fabfab-kc) [inline appeals/api/src/server/utils/horizon-gateway.js:121]: I would probably stringify this outside the if/then and reuse

**Review by fabfab-kc — CHANGES_REQUESTED** [review]
A small change for performance...

File: appeals/api/src/server/utils/horizon-gateway.js (hunk 1 above)
Comment (AlexMoulin) [reply appeals/api/src/server/utils/horizon-gateway.js:121]: Agreed

File: appeals/api/src/server/utils/horizon-gateway.js [hunk 2]
```diff
@@ -107,19 +110,27 @@ export const getAppealFromHorizon = async (caseReference) => {
 	const appealData = await got
 		.post(url, {
 			json: requestBody,
-			parseJson: (data) => parseHorizonGetCaseResponse(data)
+			parseJson: (data) => parseHorizonGetCaseResponse(data),
+			timeout: {
+				request: timeoutLimit
```
Comment (fabfab-kc) [inline appeals/api/src/server/utils/horizon-gateway.js:115]: should this be:

`request: timeoutLimit || 500`, in case is not set yet?

**Review by fabfab-kc — APPROVED** [review]
LGTM! 1 small comment but I could fast follow with the infra, so is never null...

File: appeals/api/src/server/utils/horizon-gateway.js (hunk 2 above)
Comment (fabfab-kc) [reply appeals/api/src/server/utils/horizon-gateway.js:115]: sorry just realised you had it in the config...

**Pull request merged**